core/src/main/java/org/owasp/dependencycheck/analyzer/YarnBerryAuditAnalyzer.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * This file is part of dependency-check-ant.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright (c) 2021 The OWASP Foundation. All Rights Reserved.
+ */
+package org.owasp.dependencycheck.analyzer;
+
+import org.apache.commons.collections4.MultiValuedMap;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.owasp.dependencycheck.Engine;
+import org.owasp.dependencycheck.analyzer.exception.AnalysisException;
+import org.owasp.dependencycheck.analyzer.exception.SearchException;
+import org.owasp.dependencycheck.data.nodeaudit.Advisory;
+import org.owasp.dependencycheck.dependency.Dependency;
+import org.owasp.dependencycheck.exception.InitializationException;
+import org.owasp.dependencycheck.utils.Settings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+@ThreadSafe
+public class YarnBerryAuditAnalyzer extends AbstractYarnAuditAnalyzer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(YarnBerryAuditAnalyzer.class);
+
+    @Override
+    public String getName() {
+        return "Yarn Berry Audit Analyzer";
+    }
+
+    /**
+     * Initializes the analyzer once before any analysis is performed.
+     *
+     * @param engine a reference to the dependency-check engine
+     * @throws InitializationException if there's an error during initialization
+     */
+    @Override
+    protected void prepareFileTypeAnalyzer(Engine engine) throws InitializationException {
+        super.prepareFileTypeAnalyzer(engine);
+        if (!isEnabled()) {
+            LOGGER.debug("{} Analyzer is disabled skipping yarn major version check", getName());
+            return;
+        }
+        var yarnMajorVersion = getYarnMajorVersion();
+
+        // we assume that any major version that is not the classic version, is the berry version.
+        // at the time of writing, the classic version is 1 and berry versions are 2, 3 and 4.
+        if (YARN_CLASSIC_MAJOR_VERSION < yarnMajorVersion) {
+            this.setEnabled(false);
+            LOGGER.debug("The {} has been disabled. This analyzer is not applicable for major version {}.", getName(), yarnMajorVersion);
+        }
```

Comment:
Again, I think this might have been my mistake when moving from `String` -> `int`. In the berry analyzer we want to disable the analyzer if the yarn major version is equal to the classic major version:

```suggestion
        if (YARN_CLASSIC_MAJOR_VERSION <= yarnMajorVersion) {
            this.setEnabled(false);
            LOGGER.debug("The {} has been disabled. This analyzer is not applicable for major version {}.", getName(), yarnMajorVersion);
        }
```

Reply:
Same